tests/php/FileTest.php

```diff
@@ -346,15 +348,24 @@ public function testGetURL()
     {
         /** @var File $rootfile */
         $rootfile = $this->objFromFixture(File::class, 'asdf');
-        $this->assertEquals('/assets/FileTest/55b443b601/FileTest.txt', $rootfile->getURL());
+        $this->assertEquals('/assets/55b443b601/FileTest.txt', $rootfile->getURL());
```

Comment:
I don't understand why this is no longer in `assets/FileTest` as a protected file. The `TestAssetStore` should set the root folder to `assets/FileTest`, and this particular fixture doesn't have a parent folder set. The assertion below uses `assets/FileTest` again. Is this an inconsistency in the test or the implementation?

Reply:
The Protected File Adapter and Public File Adapter have slightly different logic for generating URLs.

When you're using the TestAssetStore, protected URLs won't have the extra folder and public URLs will. I spent an hour looking at it to see if there was an elegant way to fix it, but couldn't find one. Since this only affects unit test, I decided I had bigger fish to fry.

From memory the core problems is that we can't set a different ASSET_DIR constant for our tests.

Reply:
I see. The problem is that `ProtectedAssetAdapter->getProtectedURL()` hardcodes the assets dir: https://github.com/silverstripe/silverstripe-assets/blob/1/src/Flysystem/ProtectedAssetAdapter.php#L58. The `AssetStore` which instanciates them should probably pass on its base path, rather than the adapter making guesses about `ASSETS_DIR` and `ASSETS_PATH`. But that's a pre-existing issue, and only affects tests. 

Further context from @maxime-rainville:

> The default behaviour of the various "setFileContent()" method use to be to create the file in the public store. That didn't make a lot of sense because most of the time that file would be attached to a draft entry, so the file would be move back to the protected store on save. It was also a bit problematic because natural paths don't work for versioning. So that would systematically overwrite the published file. To work around that I reverse the default. Now when in doubt we write to the protected store. A side effect of that is that some test that use to be run against the public store, are now being run against the protected one.

Reply:
See https://github.com/open-sausages/silverstripe-assets/commit/632ddceab6068a98ba2d2a178a3546b18539fb27